tests/test_cli.py

```diff
@@ -182,3 +191,169 @@ def test_find_designated_time_function(self):
+        sys.argv = [
+            "",
+            '-d', "/test_folder/",
+            '-b', "200",
+            '-e', "2018-11-18",
+            '-r',
+            '-f',
+            '-l',
+            '-g', "POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))",
+            '-t', 'test_type'
+        ]
+        with mock.patch('json.loads') as mock_json:
+            cli_copy.main()
+        self.assertIsNone(mock_json.call_args)
+
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_destination_folder_for_all_files_that_are_copied(self, mock_isfile):
+        """ the copied file(s) shall be copied at the destination folder """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
+        with mock.patch('shutil.copy') as mock_copy:
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/', src='/testing_folder/testing_file.test'),
+             call(dst='/dst_folder/', src='/new_loc_add')],
+            mock_copy.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_place_of_symlink_after_creation_of_it(self, mock_isfile):
+        """ symlink must be placed at the address that is specified from the input arguments """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-l',
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
+        with mock.patch('os.symlink') as mock_symlink:
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/testing_file.test', src='/testing_folder/testing_file.test'),
+             call(dst='/dst_folder/new_loc_add', src='/new_loc_add')],
+            mock_symlink.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('os.symlink')
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_content_of_flag_file(self, mock_isfile, mock_link):
+        """ flag file should contain this 'type: test_type' information """
+        mock_isfile.isfile.return_value = True
```

Comment:
What you are doing here is:
- you patch `geospaas_processing.cli.copy.path`, creating a Mock object
- you then edit the `isfile` property of this Mock object

This means that you are patching the whole `os.path` module, instead of just the `isfile` function, which is what you need.

So please patch `geospaas_processing.cli.copy.path.isfile` directly.

Reply:
It does not work with patching `geospaas_processing.cli.copy.path.isfile` in the decorator. We need that Mock object for the testing execution working properly. To anyway, as we have discussed internally, I go with your approach of patching.